llvm/test/tools/llvm-cas/ls-node-recursive.test

```diff
@@ -0,0 +1,116 @@
+RUN: rm -rf %t && mkdir -p %t
+RUN: clang %S/Inputs/test_a.cpp -c -gcas-friendly-debug-info --target=x86_64-apple-darwin21.1.0 -o %t/test_a.o
+RUN: clang %S/Inputs/test_b.cpp -c -gcas-friendly-debug-info --target=x86_64-apple-darwin21.1.0 -o %t/test_b.o
+RUN: find %t -name "*.o" &> %t/objects_to_ingest
+
+RUN: llvm-cas-object-format --cas %t/cas --object-stats --ingest-schema=flatv1 @%t/objects_to_ingest -silent | grep "llvmcas://" | xargs -I{}  llvm-cas --ls-node-recursive {} --cas %t/cas | FileCheck %s 
```

Comment:
Think this RUN line is over complicated. Following will do the same thing but much cleaner:
$ llvm-cas-object-format ... -silent > %toutput.casid
$ llvm-cas --ls-node-recursive @%t/output.casid | FileCheck

I know you want to use the tool to verify debug info, but I would also prefer a much simpler test here so the testcase has a clear intent for what is tested (the object visitor)

Reply:
I will change the run line, that makes sense.

Fir the test, all I am doing is checking a bunch of cas ids, it is not really doing anything specific to debug info.

Reply:
I would prefer the CAS object you used to test is a lot smaller than it currently is and you can reason about the structure easily without understanding the complicated construction schema like CASObjectFile.

Reply:
Sure, that is very do-able, thanks for the feedback